jupyterlab_nodeeditor/yggdrasil_support.py

```diff
@@ -1,9 +1,75 @@
 import uuid
-
 import yaml
 
 from .node_editor import NodeEditor
 
+import os
+import sys
+import copy
+import pytest
+import subprocess
+import itertools
+import shutil
+import logging
+from contextlib import contextmanager
+from yggdrasil.services import (
+    IntegrationServiceManager, create_service_manager_class, ServerError)
+from yggdrasil.tests import assert_raises, requires_language
+from yggdrasil import runner, import_as_function, platform
+from yggdrasil.tools import is_comm_installed, remove_path
+from yggdrasil.examples import yamls as ex_yamls
+import jupyterlab_nodeeditor as jlne
+import yggdrasil.yamlfile
+import re
+
+# Coomented out code is testing code, not sure if needed here
+
+# schema = yggdrasil.yamlfile.get_schema()
+# socket_types = tuple(schema.form_schema['definitions']['schema']['definitions']['simpleTypes']['enum'])
+# default_sockets = jlne.SocketCollection(socket_types = socket_types)
+# editor = jlne.NodeEditorModel()
+
+# photosynthesis_model = {}
+
+# # Key note here: always set the model to the actual ['model'] key of the nested dictionary
+# with open(ex_yamls['fakeplant']['python'], "r") as test_model:
+#     photosynthesis_model = yaml.safe_load(test_model)['model']
+    
+def create_slot(name, key_name, socket, socket_list, kind):
+    if kind == "i":
+        return jlne.InputSlot(title = name, key = key_name, socket_type = socket, sockets = socket_list)
+    elif kind == "o":
+        return jlne.OutputSlot(title = name, key = key_name, socket_type = socket, sockets = socket_list)
+    else:
+        return "Invalid Slot Kind"
```

Comment:
So I *think* that with the new `TraitType` implementation, you should be able to avoid a large part of this.  You should be able to supply a dictionary (there's an example notebook showing this, I believe) to the input slot attribute on a node or component, and it will construct it appropriately.